integration/airflow/openlineage/airflow/dag.py

```diff
@@ -60,44 +50,38 @@ def lineage_run_id(run_id, task, session=None):
 
     :param run_id:
     :param task:
-    :param session:
     :return:
     """
-    name = DAG._openlineage_job_name(task.dag_id, task.task_id)
-    ids = JobIdMapping.get(name, run_id, session)
-    if ids is None:
-        return ""
-    elif isinstance(ids, list):
-        return "" if len(ids) == 0 else ids[0]
-    else:
-        return str(ids)
-
-
-class DAG(airflow.models.DAG, LoggingMixin):
-    def __init__(self, *args, extractor_mapper=None, **kwargs):
+    with create_session() as session:
```

Comment:
What the benefit of using `create_session()` vs `@provide_session`?

Reply:
TBH, there's not really any. I worried that the function was being wrapped with one that held an instance of the already-created session and that the session was causing serialization issues- the user_defined_macro functions are stored as instance fields of the DAG. Looking back now, the return value of the `@provide_session` wrapper is actually just a function that only invokes `create_session` when the function is called- effectively the same as what is done here.

Reply:
Sounds good. Perhaps being more explicit to be safe isn't a bad thing 👍